### PR TITLE
Bug 1794698: Last batch of changes to be backported to 4.3

### PR DESCRIPTION
**Pull request by elfosardo**

### CODE REVIEW

**Comment by elfosardo** [comment]
talked with @stevekuznetsov  and we can safely ignore branch-ci-openshift-ironic-image-release-4.6-images failure
we'll just need an lgtm from someone in OWNERS, plus a cherry-pick-approved from a patch/release manager or the architects

File: ironic.conf [hunk 1]
```diff
@@ -20,7 +21,7 @@ deploy_logs_collect = always
 deploy_logs_local_path = /shared/log/ironic/deploy
 
 [conductor]
-automated_clean = true
+automated_clean = false
```
Comment (juliakreger) [inline ironic.conf]: This change is unrelated, but forcing it in does address a known hardware issue with IPA. This.. ideally... should be reverted once the IPA change has percolated through the release pipelines/processes.

File: ironic.conf [hunk 2]
```diff
@@ -30,7 +31,7 @@ send_sensor_data_interval = 160
 
 [deploy]
 default_boot_option = local
-erase_devices_metadata_priority = 10
+erase_devices_metadata_priority = 0
```
Comment (juliakreger) [inline ironic.conf]: This also is likely unintentional but changes the cleaning knob which addresses an issue discovered in the field on HP Hardware. We'll want to revert this line's change at some point.

**Comment by hardys** [comment]
/hold

I tested this locally with some help from @elfosardo setting up the repos to enable podman to locally build - in my tests I'm seeing the dnsmasq container crash on startup, and I needed a workaround to copy `esp.img` from `/` not `/tmp`

I think we should hold this until @elfosardo has locally tested and addressed any remaining issues

**Comment by hardys** [comment]
Not many clues from the log output:

```
[core@localhost ~]$ sudo podman logs -f dnsmasq
Waiting for ens4 interface to be configured
dnsmasq: started, version 2.79 DNS disabled
dnsmasq: compile time options: IPv6 GNU-getopt DBus no-i18n IDN2 DHCP DHCPv6 no-Lua TFTP no-conntrack ipset auth DNSSEC loop-detect inotify
dnsmasq-tftp: TFTP root is /shared/tftpboot 
/bin/rundnsmasq: line 35:    30 Killed                  sleep infinity
```

**Comment by stbenjam** [comment]
DHCP_RANGE is missing, this and https://github.com/openshift/installer/pull/3034 have to land concurrently.  That would resolve the problem.

Although, it's still odd dnsmasq is dying without DHCP_RANGE, it should just serve tftp in that case.  That's obviously not what we want, but it shouldn't die either.

**Comment by stbenjam** [comment]
We have to remove this healthcheck if there's no DHCP: https://github.com/metal3-io/ironic-image/blob/master/runhealthcheck.sh#L18

This is a bug that needs fixing

**Comment by dtantsur** [comment]
> DHCP_RANGE is missing, this and openshift/installer#3034 have to land concurrently. That would resolve the problem.

We should probably return the default for DHCP_RANGE. I hit this problem with my lab (dnsmasq did start but was not serving DHCP), spend a day trying to solve it.

**Comment by stbenjam** [comment]
> We should probably return the default for DHCP_RANGE. I hit this problem with my lab (dnsmasq did start but was not serving DHCP), spend a day trying to solve it.

We cannot do that, we must have the ability to turn off DHCP in dnsmasq for using external DHCP.

The fix for 4.3 is to ensure https://github.com/openshift/ironic-image/pull/49 and https://github.com/openshift/installer/pull/3034 land at the same time.

**Comment by juliakreger** [comment]
@dtantsur @stbenjam I believe we can now just go ahead and close this item. Are there any disagreements?

**Comment by stbenjam** [comment]
Yes that's my understanding -- :+1: for closing

**Comment by elfosardo** [comment]
as per comments above

**Pull request closed without merging**